process.py

```diff
@@ -1,119 +1,123 @@
-
 import atexit
+import datetime
 import logging
-import multiprocessing
-import operator
 import os
-import pandas as pd
+import pathlib
 import sys
-from otp4gb.batch import build_run_spec, run_batch, setup_worker
-from otp4gb.centroids import load_centroids
+
+from shapely import geometry
+
+from otp4gb.centroids import load_centroids, ZoneCentroidColumns
 from otp4gb.config import ASSET_DIR, load_config
 from otp4gb.logging import file_handler_factory, get_logger
 from otp4gb.otp import Server
-from otp4gb.util import Timer, chunker
+from otp4gb.util import Timer
+from otp4gb import cost
 
 
 logger = get_logger()
 logger.setLevel(logging.INFO)
 
-FILENAME_PATTERN = "Buffered{buffer_size}m_IsochroneBy_{mode}_ToWorkplaceZone_{location_name}_ToArriveBy_{arrival_time:%Y%m%d_%H%M}_within_{journey_time:_>4n}_mins.geojson"
+FILENAME_PATTERN = (
+    "Buffered{buffer_size}m_IsochroneBy_{mode}_ToWorkplaceZone_"
+    "{location_name}_ToArriveBy_{arrival_time:%Y%m%d_%H%M}_"
+    "within_{journey_time:_>4n}_mins.geojson"
+)
 
 
 def main():
     _process_timer = Timer()
-    matrix = []
 
     @atexit.register
     def report_time():
-        logger.info('Calculated %s rows of matrix in %s',
-                    len(matrix), _process_timer)
+        logger.info("Finished in %s", _process_timer)
 
     try:
         opt_base_folder = os.path.abspath(sys.argv[1])
-    except IndexError:
-        logger.error('No path provided')
+    except IndexError as error:
+        logger.error("No path provided")
+        raise ValueError("Base folder not provided") from error
+
     log_file = file_handler_factory(
-        'process.log', os.path.join(opt_base_folder, 'logs'))
+        "process.log", os.path.join(opt_base_folder, "logs")
+    )
     logger.addHandler(log_file)
 
     config = load_config(opt_base_folder)
 
-    opt_travel_time = config.get('travel_time')
-    opt_buffer_size = config.get('buffer_size', 100)
-    opt_modes = config.get('modes')
-    opt_centroids_path = os.path.join(ASSET_DIR, config.get('centroids'))
-    opt_clip_box = operator.itemgetter(
-        'min_lon', 'min_lat', 'max_lon', 'max_lat')(config.get('extents'))
-
-    opt_isochrone_step = 15
-    opt_max_travel_time = 90
-    opt_max_walk_distance = 2500
-    opt_no_server = False
-    opt_num_workers = 8
-    opt_chunk_size = opt_num_workers * 10
-    opt_name_key = 'msoa11nm'
+    opt_centroids_path = os.path.join(ASSET_DIR, config.centroids)
 
     # Start OTP Server
     server = Server(opt_base_folder)
-    if not opt_no_server:
-        logger.info('Starting server')
+    if not config.no_server:
+        logger.info("Starting server")
         server.start()
 
     # Load Northern MSOAs
```

Comment:
Load northern boundaries? (they don't have to be MSOA?)

Reply:
Fixed at some point, can't remember when